scripts/build/Jenkins/Jenkinsfile

```diff
@@ -338,6 +338,11 @@ def PreBuildCommonSteps(Map pipelineConfig, String snapshot, String repositoryNa
         else command += '.cmd'
         command += " -u ${pipelineConfig.BUILD_ENTRY_POINT} --platform ${platform} --type clean"
         palSh(command, "Running ${platform} clean")
+
+        if(fileExists('.lfsconfig')) {
+            palSh("git lfs install", "LFS config exists. Installing LFS hooks to local repo")
+            palSh("git lfs pull", "Pulling new LFS objects")
+        }
```

Comment:
we had a code similar to this that was removed at some point 
@brianherrera do you remember why?

Reply:
Yes, I recall @brianherrera removed it here: https://github.com/o3de/o3de/pull/2097

Reply:
Yes this was removed because we no longer required a separate lfs pull with the public repos which require additional auth step as well.

Reply:
This should be fine to put back in since this no longer requires a separate auth step which was a potential risk to expose credentials.

@amzn-changml does this issue occur when the lfs filter is updated with new file types?

Reply:
It seems to be related to either of these two situations, but not directly related to changing the filetype:
- An LFS object is deleted with `git rm` and later readded
- An LFS object is renamed